### ISSUE TITLE
default_trait_access False Positive with TypedBuilder

### ISSUE THREAD

**Issue by simonsan** [rust-lang/rust-clippy#6858]
Lint name:
`clippy::default_trait_access`

I tried this code:

```rust
error: calling `std::option::Option::default()` is more clear than this expression
   --> src\domain\types\api\match_info_response.rs:233:24
    |
233 | #[derive(Clone, Debug, TypedBuilder, PartialEq, Serialize, Deserialize)]
    |                        ^^^^^^^^^^^^
    |
note: the lint level is defined here
   --> src/lib.rs:3:9
    |
3   | #![deny(clippy::pedantic)]
    |         ^^^^^^^^^^^^^^^^
    = note: `#[deny(clippy::default_trait_access)]` implied by `#[deny(clippy::pedantic)]`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#default_trait_access
    = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
```

I expected to see this happen: 
`To not show up`

Instead, this happened: See above

### Meta

- `cargo clippy -V`: clippy 0.1.52 (45b3c285 2021-03-04)
- `rustc -Vv`:
  ```
  rustc 1.50.0 (cb75ad5db 2021-02-10)
  binary: rustc
  commit-hash: cb75ad5db02783e8b0222fee363c5f63f7e2cf5b
  commit-date: 2021-02-10
  host: x86_64-pc-windows-msvc
  release: 1.50.0
  ```
